ExampleCode/C++/API/API.cpp

```diff
@@ -413,6 +483,114 @@ bool iSENSE::get_project_fields()
 }
 
 
+// Given that a project ID has been set, this function
+// makes a GET request and saves all the datasets & media objects
+// into two picojson arrays.
+// It will also update the fields for that given project ID
+bool iSENSE::get_datasets_and_mediaobjects()
+{
+  // Check that the project ID is set properly.
+  // When the ID is set, the fields are also pulled down as well.
+  if(project_ID == "empty" || project_ID.empty())
+  {
+    cout << "\nError - please set a project ID!\n";
+    return false;
+  }
+
+  // The "?recur=true" will make iSENSE return:
+  // ALL datasets in that project.
+  // ALL media objects in that project
+  // And owner information.
+  get_URL = devURL + "/projects/" + project_ID + "?recur=true";
+
+  // This project will try using CURL to make a basic GET request to rSENSE
+  // It will then save the JSON it recieves into a picojson object.
+  CURL *curl = curl_easy_init();      // cURL object
+  long http_code;                     // HTTP status code
+  MEMFILE* json_file = memfopen();    // Writing JSON to this file.
+  char error[256];                    // Errors get written here
+
+  if(curl)
+  {
+    curl_easy_setopt(curl, CURLOPT_URL, get_URL.c_str());
+    curl_easy_setopt(curl, CURLOPT_ERRORBUFFER, &error);    // Write errors to the array "error"
+
+    // From the picojson example, "github-issues.cc". Used  for writing the JSON to a file.
+    curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, memfwrite);
+    curl_easy_setopt(curl, CURLOPT_WRITEDATA, json_file);
+
+    // Perform the request, res will get the return code
+    curl_easy_perform(curl);
+
+    // We can actually get the HTTP response code from cURL, so let's do so to check for errors.
+    http_code = 0;
+
+    // This will put the HTTP response code into the "http_code" variable.
+    curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &http_code);
+
+    /*
+     *  The iSENSE API gives us one response code to check against:
+     *  Success: 200 OK
+     *  Failure: 404 Not Found
+     */
+
+    // If we do not get a code 200, or cURL quits for some reason, we didn't successfully
+    // get the project's fields.
+    if(http_code != 200)
+    {
+      cout << "\nGET project fields failed.\n";
+      cout << "Is the project ID you entered valid?\n";
+
+      // Clean up cURL and close the memfile
+      curl_easy_cleanup(curl);
+      curl_global_cleanup();
+      memfclose(json_file);
+
+      return false;
+    }
+
+    // We got a code 200, so try and parse the JSON into a PICOJSON object.
+    // Error checking for the parsing occurs below.
+    string errors;
+
+    // This will parse the JSON file.
+    parse(get_data, json_file->data, json_file->data + json_file->size, &errors);
+
+    // If we have errors, print them out and quit.
+    if(errors.empty() != true)
+    {
+      cout << "\nError parsing JSON file!\n";

```

Comment:
Error parsing JSON file... in which method?  May want to include the method name in this kind of output.